src/epggrab/otamux.c

```diff
@@ -673,13 +673,26 @@ epggrab_ota_kick_cb ( void *p )
                                 SUBSCRIPTION_TABLES))) {
     if (r != SM_CODE_NO_ADAPTERS) {
       tvhtrace(LS_EPGGRAB, "subscription failed for %s (result %d)", mm->mm_nicename, r);
-      TAILQ_INSERT_TAIL(&epggrab_ota_pending, om, om_q_link);
-      om->om_q_type = EPGGRAB_OTA_MUX_PENDING;
-      om->om_retry_time = mclk() + mono2sec(60);
-      if (r == SM_CODE_NO_FREE_ADAPTER)
-        net->failed = 1;
-      if (r == SM_CODE_TUNING_FAILED)
+      if( om->om_retry_count < 3 ) {
+        TAILQ_INSERT_TAIL(&epggrab_ota_pending, om, om_q_link);
+        om->om_q_type = EPGGRAB_OTA_MUX_PENDING;
+        if (r == SM_CODE_NO_FREE_ADAPTER) {
+          /* wait for scheduled mux to finish OTA grab */		 
+          om->om_retry_time = mclk() + sec2mono(60);
+	  om->om_retry_count = 0;
+	}
+	else {
+          /* cannot tune - increase retry count and try again later */		 
+          om->om_retry_time = mclk() + sec2mono(30);
+	  ++(om->om_retry_count);
```

Comment:
Why not simply `om->om_retry_count++;`? Also intendation is not correct here

Reply:
old habit from a grey beard coder i guess. Pre-increment has traditionally been considered more code efficient than post, but that matter little nowadays on modern compilers - I can fix that if this is the expected syntax for simple increment.

Reply:
Everywhere in the file post-increment is used and I think it improves readability. With all the crazy optimizations going on it will probably end up as the same instructions anyways. I'm super sorry that I didn't respond before you fixed the other things and squashed your commits :/

Reply:
I'll fix it for sure if this is the last change requested, but i have doubts somehow based on later comments  😊

Reply:
just fixed the increment